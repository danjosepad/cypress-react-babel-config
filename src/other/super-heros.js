const superHeros = [
  {name: 'Dynaguy', powers: ['desintegration ray', 'fly']},
  {name: 'Apogee', powers: ['gravity control', 'fly']},
  {name: 'Blazestone', powers: ['control of fire', 'pyrothecnic discharges']},
  {name: 'Frozone', powers: ['freeze water']},
  {name: 'Mr. Incredible', powers: ['physical strength']},
  {name: 'Elastigirl', powers: ['physical stretch']},
  {name: 'Violet', powers: ['invisibility', 'force fields']},
  {name: 'Dash', powers: ['speed']},
]

function getFlyingSuperHeros(){
  return superHeros.filter(hero => {
    return hero.powers.includes('fly')
  })
}

export {getFlyingSuperHeros}