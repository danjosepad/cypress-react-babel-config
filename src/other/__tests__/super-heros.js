import { getFlyingSuperHeros } from '../super-heros'

test('returns super heros that can fly', () => {
  const flyingHeros = getFlyingSuperHeros()
  expect(flyingHeros).toEqual([
    { name: 'Dynaguy', powers: [ 'desintegration ray', 'fly' ] },
    { name: 'Apogee', powers: [ 'gravity control', 'fly' ] }
  ])
})