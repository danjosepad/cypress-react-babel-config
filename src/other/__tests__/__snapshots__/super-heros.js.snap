// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns super heros that can fly 1`] = `
Array [
  Object {
    "name": "Dynaguy",
    "powers": Array [
      "desintegration ray",
      "fly",
    ],
  },
  Object {
    "name": "Apogee",
    "powers": Array [
      "gravity control",
      "fly",
    ],
  },
]
`;
